pkg/stanza/fileconsumer/splitter_factory.go

```diff
@@ -25,16 +25,21 @@ type splitterFactory interface {
 }
 
 type multilineSplitterFactory struct {
-	helper.SplitterConfig
+	EncodingConfig     helper.EncodingConfig
+	Flusher            helper.FlusherConfig
+	Multiline          helper.MultilineConfig
+	PreserveWhitespace bool
```

Comment:
This PR looks good to me and I'm ok with merging it, but I had a thought that we should decide on first.

Do you think we might eventually have a need to support similar options, like `preserve_leading` and `preserve_trailing`?

If so, this boolean would probably need to be deprecated in favor of either separate booleans or an enumeration. 

I think two booleans could easily be evaluated together to choose the correct trimFunc, and otherwise the processing remains the same.

Should we just do this now to protect against a breaking change later? Up to you.

Reply:
I am not sure if there is need to only trim the leading or trailing whitespace in future, I think maybe having this only option to trim or keep them both for now, and in future if necessary moving the trim functionality to an operator makes more sense?

Reply:
Either way sounds fine to me. Typically we try to define boolean parameters so that they default to false, but I do think the names you suggested are better, so seems like a reasonable exception.

Reply:
It looks like having parameters default to true causing issue in `syslog_input` [config](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/operator/input/syslog/syslog.go#L51-L52), which supports embedded `tcp_input` and `udp_input`, any suggestion, maybe we need to stick to false as default values and rename parameters as presevingXxxWhitespaces?

from https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4251161435/jobs/7393183635#logs

```
--- FAIL: TestUnmarshal (0.01s)
    --- PASS: TestUnmarshal/default (0.00s)
    --- FAIL: TestUnmarshal/tcp (0.00s)
    --- FAIL: TestUnmarshal/udp (0.00s)
```

Reply:
Hmm, seems to be one more reminder why defaulting to false is preferred. :)

> maybe we need to stick to false as default values and rename parameters as presevingXxxWhitespaces

👍 Let's go with this.